src/power_grid_model/core/data_handling.py

```diff
@@ -101,7 +102,7 @@ def prepare_output_view(output_data: Mapping[ComponentType, np.ndarray], output_
 
 
 def create_output_data(
-    output_component_types: Union[Set[ComponentType], List[ComponentType]],
+    output_component_types: OutputComponentNamesType,
```

Comment:
sparse batch data is not captured by this ticket, correct?

Reply:
output would only be dense data.

Reply:
Does it make sense for the user to provide only components for the output but ask them to be in `ColumnarData` type? If yes, the output would be of sparse data, right?

Reply:
Sparse/dense is only the case for now and may change in the future if we start applying filtering by e.g. id as well

Reply:
Use case is when a user is only interested in a specific id or only in energized components